Controller/Adminhtml/Query/Save.php

```diff
@@ -84,9 +93,6 @@ public function execute()
 
             if (isset($data['banner_image'][0]['name']) && isset($data['banner_image'][0]['tmp_name'])) {
                 $data['banner_image'] = $data['banner_image'][0]['name'];
-                $this->imageUploader = \Magento\Framework\App\ObjectManager::getInstance()->get(
-                    'Algolia\AlgoliaSearch\QueryImageUpload'
-                );
                 $this->imageUploader->moveFileFromTmp($data['banner_image']);
```

Comment:
Hello @vmalyk . For some reason, with this change, it breaks on save with this errror : 

`Exception #0 (BadMethodCallException): Missing required argument $baseTmpPath of Algolia\AlgoliaSearch\Model\ImageUploader.`

Seems that the imageUploader has to be defined in the di.xml as it's done for [ImageUpload](https://github.com/algolia/algoliasearch-magento-2/blob/develop/etc/di.xml#L97-L101) .

When I add the same lines below line 101, for  
`<type name="Algolia\AlgoliaSearch\Controller\Adminhtml\Query\Save">`,
the error seems to disappear.

Reply:
@damcou , yes, it`s right. I added to di.xml injection ImageQueryUploader